### PR TITLE
Fix problem with duplicate groupId:ArtifactId.

### PR DESCRIPTION
**Pull request by Bimeche**
Solved by using ManagementKey
Fixes issue #96

### CODE REVIEW

File: src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java [hunk 1]
```diff
@@ -880,7 +880,7 @@ private static void updateDependencyVersions(Element element, ModelBase model) {
     private static void updateDependencyVersions(Element dependenciesElement, List<Dependency> dependencies) {
         final Map<String, String> dependencyVersionMap = dependencies.stream()
                 .filter(it -> it.getVersion() != null)
-                .collect(toMap(it -> it.getGroupId() + ":" + it.getArtifactId(), Dependency::getVersion));
+                .collect(toMap(Dependency::getManagementKey, Dependency::getVersion));
```
Comment (qoomon) [inline src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java:883]: I think we need to adjust line 890 as well, don't we?

**Comment by qoomon** [comment]
Hi @Bimeche, thanks for the report. May you can upload a simple example project with that specific exception?

File: src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java (hunk 1 above)
Comment (Bimeche) [reply src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java:883]: Changing only line 883 fixed the issue for me, so I didn't really look more into it, but now that I do, it seems like a better fix yes.
So something like
```
String dependencyGroupId = dependencyElement.getChild("groupId").getText();
String dependencyArtifactId = dependencyElement.getChild("artifactId").getText();
String dependencyType = dependencyElement.getChild("type").getText();
String dependencyClassifier = dependencyElement.getChild("classifier") != null ? dependencyElement.getChild("classifier").getText() : null;
Element dependencyVersionElement = dependencyElement.getChild("version");
if (dependencyVersionElement != null) {
    dependencyVersionElement.setText(dependencyVersionMap.get(dependencyGroupId + ":" +
                                                              dependencyArtifactId + ":" +
                                                              dependencyType +
                                                              (dependencyClassifier != null ? ":" + dependencyClassifier : "")));
```
to mimic the getManagementKey method ?

Comment (Bimeche) [reply src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java:883]: Here is a test project for this bug.
https://github.com/Bimeche/TestExtension

**Comment by qoomon** [comment]
I did some more refactoring. I got rid of creating and using a map at all. 

Thx again for reporting.

I will release a new version `6.2.1` within the next couple of hours.

**Pull request merged**